### ISSUE TITLE
Test for mis-formatted text

### ISSUE THREAD

**Issue by bdarfler** [olafurpg/scalafmt#114]
Would love (specifically in SBT but probably in other contexts) to have an option to test for formatting violations (so we can fail a build) without having to actually format the code.

**Comment by olafurpg** [comment]
I agree, want this for my project too. Will try to fit this in the 0.2 release, gonna iron out some output bugs first.

**Comment by bdarfler** [comment]
:+1:

**Comment by olafurpg** [comment]
So I understand correctly, there should be an sbt command like `scalafmtFormatTest` that exits the process with status 1 in case it finds mis-formatted code in the project?

**Comment by sjrd** [comment]
An sbt task/command should throw an exception on failure, instead of exiting the process. But yes, otherwise that's what I would like.

**Comment by bdarfler** [comment]
Seconded. Simply throwing an exception would be a great start. Longer term it might be nice to give details on what wasn't formatted correctly but really the point is to run the formatter so that's more of a nice to have.

**Comment by olafurpg** [comment]
Got it. I have a working implementation, which I hope to merge for the 0.2.0 release later this week (if all goes as planned).

**Comment by olafurpg** [comment]
The task `scalafmtTest` is part of the 0.2.0 release, which contains many other goodies. Docs are [here](https://olafurpg.github.io/scalafmt/#SBT).

**Comment by bdarfler** [comment]
:clap: Thanks!

**Comment by bdarfler** [comment]
Running scalafmtTest in 0.2.1 I'm seeing this

```
[warn] Scalafmt error, please report to https://github.com/olafurpg/scalafmt/issues
[warn]   /Users/Bdarfler/code/reservoir/src/main/scala/com/localytics/reservoir/Values.scala:
[warn]   /Users/Bdarfler/code/reservoir/src/main/scala/com/localytics/reservoir/Values.scala is mis-formatted.
```

I expected that this would fail the build instead of just printing a warning.

**Comment by olafurpg** [comment]
I know what the issue is. I changed 0.2.1 to throw an exception instead of system.exit in 0.2.0. I forgot I catch all non fatal exceptions higher above so the error is swallowed. I'll fix it for 0.2.2

**Comment by bdarfler** [comment]
Thanks! BTW, I can't tell you how excited I am for this! Thanks!
